module/data/item/weapon.mjs

```diff
@@ -246,6 +246,39 @@ export default class WeaponData extends ItemDataModel.mixin(
 
   /* -------------------------------------------- */
 
+  /** @override */
+  get attackModes() {
+    const modes = [];
+
+    // Thrown ranged weapons will just display the "Thrown" mode
+    if ( !(this.properties.has("thr") && (this.attackType === "ranged")) ) {
+      // Weapons without the "Two-Handed" property or with the "Versatile" property will have One-Handed attack
+      if ( this.isVersatile || !this.properties.has("two") ) modes.push({
+        value: "oneHanded", label: game.i18n.localize("DND5E.ATTACK.Mode.OneHanded")
+      });
+
+      // Weapons with the "Two-Handed" property or with the "Versatile" property will have Two-Handed attack
+      if ( this.isVersatile || this.properties.has("two") ) modes.push({
+        value: "twoHanded", label: game.i18n.localize("DND5E.ATTACK.Mode.TwoHanded")
+      });
+    }
+
+    // Weapons with the "Light" property will have Offhand attack
+    if ( this.properties.has("lgt") ) modes.push({
```

Comment:
Hmm, I'd prefer not to add it to all users since most users can't use it. Maybe we can just take this opportunity to add an Enhanced Duel Wielding flag to handle that feat.

Reply:
I just don't see the harm in restricting it. If it's not applicable they can just... not select it, right? We use this philosophy throughout the system.